azure/batch/scripts/predict_on_catalog.py

```diff
@@ -104,39 +106,107 @@ def setup(self, stage: Optional[str] = None):
 
 def save_predictions_to_json(predictions, id_str, label_cols, save_loc):
     # JSON output format is used for services like the zooniverse subject assistant
+    # Could add any other decision rules into this function
     assert save_loc.endswith('.json')
-    # setup the output data structur with a schema describing the data
+    # setup the output data structure with a schema describing the data
     output_data = {
       'schema': {
         'version': 1,
         'type': 'zooniverse/subject_assistant',
-        'data': {'subject_id': 'probability_galaxy_is_not_smooth'}
+        'data': {
+            'predictions': {'subject_id': ['smooth-or-featured-cd_smooth', 'smooth-or-featured-cd_featured-or-disk', 'smooth-or-featured-cd_problem']},
+            'probabilities': {'subject_id': 'probability_at_least_20pc_featured'}
+        }
       }
+      # will add the actual data under 'data' key, below
     }
+
+    # check that the predictions mean what we think they mean
+    assert label_cols[0] == 'smooth-or-featured-cd_smooth'
+    assert label_cols[1] =='smooth-or-featured-cd_featured-or-disk'
+    assert label_cols[2] == 'smooth-or-featured-cd_problem'
```

Comment:
i'm ok to leave these in instead of outright failures - can we capture the assert failures and warn / critical in the logs about the type of failure (data corruption) this is leading too... happy to re-raise the failure and stop processing as well

I think in terms of maintenance that type of log output will massively help any future developer understand the error better (i.e. the inputs have changed and are not for the questions we expect).